test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -11889,14 +11875,15 @@ def add_dedicated_time(self, as_str=None, start=None, end=None, hup=True):
         if dtime_from is not None and dtime_to is not None:
             self.dedicated_time.append({'from': dtime_from, 'to': dtime_to})
         try:
-            (fd, fn) = self.du.mkstemp()
+            fn = self.du.create_temp_file()
+            fd = open(fn, "w")
```

Comment:
Same as above, consider using a with statement (and for any other occurrences like this)